.github/workflows/build-docker.yml

```diff
@@ -0,0 +1,60 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+name: Build Docker Image
+
+on:
+  release:
+    types: [ published ]
+
+env:
+  dotnetVersion: "6.0"
+
+jobs:
+  build-docker:
+    name: Build & Push Docker Image
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login to GHCR
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.PAT }}
+
+      - name: Extract Image Metadata
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ github.repository }},ghcr.io/${{ github.repository }}
+          tags: |
+            type=semver,pattern={{major}}.{{minor}}.{{patch}},value=${{ github.event.release.tag_name }}
+            type=semver,pattern={{major}}.{{minor}},value=${{ github.event.release.tag_name }}
+            type=semver,pattern={{major}},value=${{ github.event.release.tag_name }}
+
+      - name: Build & Push Image
```

Comment:
I think one thing we'll need is to split the build & publish into two steps. If people modify the `Dockerfile` in the future, we probably want to run that build every time a commit happens that touches those files. That way, we can catch any build issues early. Publishing only should happen each time a new release happens (for now).

I know you've already done a lot of work, so I don't want to push more on you. If you don't want to or can't get to this, I'll find some time later on to copy your changes to a branch and make those amendments. But I'd really like to make sure that folks (myself included) don't accidentally break the docker build!

Reply:
Maybe I'll find some time tomorrow to do that. I think this should be as simple as adding a trigger for PRs/Pushes and setting the `push` input of the step to something like `${{ github.event_name == 'release' }}`, but I'll have to test it...

Reply:
I'm trying to wrap up some major changes in Sonarr support for JSON files in the guide. Once I get done with that, I'll help you out. I have too much to work on! :(